source/common/secret/secret_manager_impl.h

```diff
@@ -50,7 +50,8 @@ class SecretManagerImpl : public SecretManager {
     findOrCreate(const envoy::api::v2::core::ConfigSource& sds_config_source,
                  const std::string& config_name,
                  Server::Configuration::TransportSocketFactoryContext& secret_provider_context) {
-      const std::string map_key = sds_config_source.SerializeAsString() + config_name;
+      const std::string map_key =
```

Comment:
You can just use uint64_t for map_key type and change type elsewhere, no need convert to string.

Reply:
since resourcename is also needed to construct mapkey, are you suggesting using something like ```MessageUtil::hash(sds_config_source+config_name)``` ?

Reply:
Ah sorry I meant rds_impl

Reply:
done.